### PR TITLE
refactor threadline widget

### PR DESCRIPTION
**Pull request by pazz**
This simplifies the building of a threadline widget in search mode.

This branch also includes (refactored) variants of #1261 and #1262

### CODE REVIEW

File: alot/widgets/search.py [hunk 1]
```diff
@@ -175,11 +79,146 @@ def keypress(self, size, key):
...
-            path.append('focus')
+
+def build_tags_part(tags, attr_normal, attr_focus):
+    """
+    create an urwid.Columns widget (wrapped in approproate Attributes)
+    to display a list of tag strings, as part of a threadline.
+
+    :param tags: list of tag strings to include
+    :type tags: list of str
+    :attr_normal: urwid attribute to use if unfocussed
```
Comment (dcbaker) [inline alot/widgets/search.py]: :param attr_normal:

File: alot/widgets/search.py [hunk 2]
```diff
@@ -175,11 +79,146 @@ def keypress(self, size, key):
...
+    :attr_focus: urwid attribute to use if focussed
+    :return: overall width in characters and a Columns widget.
+    :rtype: int, :class:`urwid.Columns`
```
Comment (dcbaker) [inline alot/widgets/search.py]: I think this should be `:rtype: tuple[int, urwid.Columns]`

File: alot/widgets/search.py [hunk 3]
```diff
@@ -175,11 +79,146 @@ def keypress(self, size, key):
...
+
+    :param name: id of part to build
+    :type name: str
+    :param thread: the thread to get local info for
+    :type thread: :class:`alot.db.thread.Thread`
+    :param struct: theming attributes for this part, as provided by
+                   :class:`alot.settings.theme.Theme.get_threadline_theming`
+    :type struct: dict
+    :return: overall width (in characters) and a widget.
+    :rtype: int, :class:`AttrFlipWidget`
```
Comment (dcbaker) [inline alot/widgets/search.py]: here too, `tuple[int, AttrFliwWidget]`

**Comment by pazz** [comment]
done, cheers!

**Pull request merged**